src/message_pack/from_msgpack.cr

```diff
@@ -110,7 +110,7 @@ def Union.new(pull : MessagePack::Unpacker)
         return {{type}}.new(pull) if token.is_a?(MessagePack::Token::IntT)
       {% elsif type == Float32 || type == Float64 %}
         return {{type}}.new(pull) if token.is_a?(MessagePack::Token::FloatT)
-        {% unless T.any? { |t| t.name.starts_with?("Int") } %}
+        {% unless T.any? { |t| t < Int } %}
```

Comment:
what this `<` means? is there only types or what?

Reply:
When comparing types it checks inheritance. In this case all int types are subtypes of Int. Docs [here](https://crystal-lang.org/api/0.34.0/Crystal/Macros/TypeNode.html#%3C(other:TypeNode):BoolLiteral-instance-method)